libraries/AP_HAL_ChibiOS/hwdef/CUAV-Nora/hwdef.dat

```diff
@@ -0,0 +1,311 @@
+PE14 TIM1_CH4  TIM1  PWM(12) GPIO(61)
+PH6  TIM12_CH1 TIM12 PWM(13) GPIO(62) NODMA
+PH9  TIM12_CH2 TIM12 PWM(14) GPIO(63) NODMA
+
+# WS2812 LED
+#PI5 TIM8_CH1 TIM8 PWM(15) GPIO(64)
+
+# PWM output for buzzer
+PE5 TIM15_CH1 TIM15 GPIO(77) ALARM
+PB4 TIM3_CH1 TIM3 RCININT PULLUP LOW 
+
+# analog in
+PA0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PA1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+# ADC3.3/ADC6.6
+PC4 SPARE1_ADC1 ADC1 SCALE(1)
+PA4 SPARE2_ADC1 ADC1 SCALE(1)
+
+PF12 RSSI_IN ADC1 SCALE(1)
+
+PC5 VDD_5V_SENS ADC1 SCALE(2)
+PC1 SCALED_V3V3 ADC1 SCALE(2)
+
+# CAN bus
+PI9  CAN1_RX CAN1
+PH13 CAN1_TX CAN1
+
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+PH2 GPIO_CAN1_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(70)
+PH3 GPIO_CAN2_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(71)
+
+# GPIOs
+
+PA8 HEATER_EN OUTPUT LOW GPIO(80)
+define HAL_HEATER_GPIO_PIN 80
+define HAL_HAVE_IMU_HEATER 1
+
+PG1  VDD_BRICK_VALID INPUT PULLUP
+PG2  VDD_BRICK2_VALID INPUT PULLUP
+PA9  VBUS INPUT
+PJ3  VDD_5V_HIPOWER_OC INPUT PULLUP
+PJ4  VDD_5V_PERIPH_OC INPUT PULLUP
+PB10 nSPI5_RESET_EXTERNAL1 OUTPUT HIGH
+
+# SPI devices
+SPIDEV rm3100         SPI1 DEVID1  RM3100_CS       MODE3  2*MHZ  8*MHZ
+SPIDEV icm20689       SPI1 DEVID2  ICM20689_CS     MODE3  2*MHZ  8*MHZ
+SPIDEV bmi088_a       SPI4 DEVID2  BMI088_A_CS     MODE3  10*MHZ 10*MHZ
+SPIDEV bmi088_g       SPI4 DEVID3  BMI088_G_CS     MODE3  10*MHZ 10*MHZ
+SPIDEV ms5611_imu     SPI4 DEVID1  MS5611_IMU_CS   MODE3  20*MHZ 20*MHZ
+SPIDEV ms5611_board   SPI6 DEVID1  MS5611_BOARD_CS MODE3  20*MHZ 20*MHZ
+SPIDEV icm20649       SPI6 DEVID2  ICM20649_CS     MODE3  2*MHZ  8*MHZ
+SPIDEV ramtron        SPI2 DEVID1  FRAM_CS         MODE3  8*MHZ  8*MHZ 
+
+# two baro
+BARO MS56XX SPI:ms5611_imu
+BARO MS56XX SPI:ms5611_board
+
+# three IMUs
+IMU Invensense   SPI:icm20689 ROTATION_NONE
+IMU BMI088 SPI:bmi088_a SPI:bmi088_g ROTATION_PITCH_180_YAW_270
+IMU Invensensev2 SPI:icm20649 ROTATION_NONE
+
+# compasses
+
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+COMPASS IST8310 I2C:ALL_EXTERNAL:0x0E true  ROTATION_ROLL_180_YAW_90
+COMPASS IST8310 I2C:ALL_INTERNAL:0x0E false ROTATION_ROLL_180_YAW_90
+
+COMPASS RM3100 SPI:rm3100 false ROTATION_NONE 
+ 
+
+# microSD support (disabled for now)
+PC8 SDMMC1_D0 SDMMC1
+PC9 SDMMC1_D1 SDMMC1
+PC10 SDMMC1_D2 SDMMC1
+PC11 SDMMC1_D3 SDMMC1
+PC12 SDMMC1_CK SDMMC1
+PD2 SDMMC1_CMD SDMMC1
+
+# red LED marked as B/E
+
+PD8 LED_R1 OUTPUT OPENDRAIN HIGH GPIO(0)
+PC6 LED_G1 OUTPUT OPENDRAIN LOW GPIO(1)
+PC7 LED_B1 OUTPUT OPENDRAIN HIGH GPIO(2)
+
+define HAL_GPIO_A_LED_PIN 0
+define HAL_GPIO_B_LED_PIN 1
+define HAL_GPIO_C_LED_PIN 2
+
+define HAL_GPIO_LED_ON  0
+define HAL_GPIO_LED_OFF 1
+
+define HAL_HAVE_PIXRACER_LED
+
+# enable RAMTROM parameter storage
+define HAL_STORAGE_SIZE 32768
+define HAL_WITH_RAMTRON 1
+
+# allow to have have a dedicated safety switch pin
+define HAL_HAVE_SAFETY_SWITCH 1
+PE12 LED_SAFETY OUTPUT
+PE10 SAFETY_IN INPUT PULLDOWN
+
+# also probe for external compasses
+
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
```

Comment:
these 3 compass lines are not needed here

Reply:
![image](https://user-images.githubusercontent.com/30532769/72672644-116cbb80-3a98-11ea-87cf-69fe74530e5f.png)
Don't need to enable external compass detection here, isn't  ?

Reply:
you had it in the file twice, you only need it once